tt_eager/tensor/tensor.cpp

```diff
@@ -406,9 +406,8 @@ Tensor Tensor::cpu_sharded() const {
 
 
 Tensor Tensor::extract_shard(const CoreCoord & core) const{
-
-    auto buffer= this->buffer();
-    uint32_t core_id = buffer->core_to_core_id().at(core);
+    auto buffer_page_mapping = generate_buffer_page_mapping(*this->buffer());
+    auto core_id = buffer_page_mapping.core_to_core_id_.at(core);
     return this->extract_shard(core_id);
 }
 
```

Comment:
Is it possible to make all these standalone function and outside of the Tensor class?

Reply:
This is not something that needs to be done on this PR.  Just something we want to drive to.